src/components/Profile.svelte

```diff
@@ -1,17 +1,104 @@
 <script>
+  import { db } from "../utils/api";
   import { handleLogout } from "../utils/auth";
-  import { useNavigate, useLocation } from "svelte-navigator";
   import { user } from "../utils/stores";
+  import { getAuth, updatePassword, onAuthStateChanged } from "firebase/auth";
+  import {
+    onSnapshot,
+    collection,
+    deleteDoc,
+    doc,
+    updateDoc,
+  } from "firebase/firestore";
+  import { update } from "firebase/database";
+  import UpdateListings from "./UpdateListings.svelte";
+
+  const auth = getAuth();
+
+  let fireUser;
+
+  onAuthStateChanged(auth, (firebaseUser) => {
+    if (firebaseUser) {
```

Comment:
is onAuthStateChanged needed specially instead of the user.subscribe context and signedIn variable below which gets current user ID? Don't know if it's something particular for the change password function, If not then not needed.

Reply:
Yeah we because of the async nature, this was the only way to get the info we needed